llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -200,6 +200,11 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
 
   setOperationAction(ISD::UADDO, isPPC64 ? MVT::i64 : MVT::i32, Custom);
 
+  // On P10, the default lowering generates better code using the
+  // setbc instruction.
+  if (!Subtarget.hasP10Vector() && isPPC64)
```

Comment:
Why only for 64-bit?

Reply:
I  [pre committed](https://github.com/llvm/llvm-project/commit/66d350a01738c0c64a28b943af5e995a5fbc348d) a test and will enable this for 32-bit, thanks.